### DISCUSSION TITLE
Svelte-cms and AWS for storage

### DISCUSSION THREAD

**Discussion by marksteven** [sveltia/sveltia-cms#21] [General]
Currently using Sveltekit and AWS for storage (s3) Lambda serverless functions, cognito authenticaion.
Was interested to see  if svelteia-cms could be possibly used on a AWS platform?

**Comment by kyoshino** [comment]
Thank you for checking out Sveltia CMS! Hmm, authentication is the tricky part.

I can find a compatible [authentication script for Lambda](https://github.com/marksteele/netlify-serverless-oauth2-backend) (or you can use [our script for Cloudflare Workers](https://github.com/sveltia/sveltia-cms-auth); it doesn’t have to be on the same platform) but it authenticates with GitHub, not Cognito. The users have to be managed with a GitHub repository’s access control.

How many users do you have? If there are only a few, let them sign up with GitHub, then you can invite them to your repository. If there are many and/or if your repository is on CodeCommit, you might be able to write your own auth script. Not sure.

Either way, Sveltia CMS nor Netlify/Decap CMS doesn’t have [roles](https://github.com/decaporg/decap-cms/issues/2) yet. All users have the same permission to create/delete entries.

Let me know if there is anything I can help with!

**Reply by kyoshino** [reply]
> if your repository is on CodeCommit, you might be able to write your own auth script.

Well, no. Sveltia CMS doesn’t support CodeCommit. Your repo has to be on GitHub.

**Reply by mnordhaus** [reply]
Code Commit is discontinued:
https://aws.amazon.com/de/blogs/devops/how-to-migrate-your-aws-codecommit-repository-to-another-git-provider/
